ScarletEngine/Sources/Engine/Public/Engine.h

```diff
@@ -26,7 +28,7 @@ namespace ScarletEngine
 		/** Remove a tickable object. This must be immediate as it will only be called on destruction of an ITickable */
 		void RemoveTickable(ITickable* TickableObject);
 
-		void Terminate();
+		ApplicationWindow* GetApplicationWindow() const { return AppWindow; }
```

Comment:
Allowing anyone to get a pointer to the window seems kind of weird, would it be possible to have the Engine object pass this pointer to whatever system needs it? If not it's whatever, just curious

Reply:
Yeah I agree I don't like how it is now but I'm doing it this way just to save some time implementing the interface allowing access to window events. I'll have to refactor this later for sure to pipe the events through a more abstract interface.